web/src/inbox_ui.js

```diff
@@ -462,7 +462,10 @@ export function complete_rerender() {
     );
     update_filters();
     show_empty_inbox_text(has_visible_unreads);
-
+    // If the focus is not on the inbox rows, inbox scrolls down when moving
+    // from other narrows to inbox view. To avoid it we scroll to top before
+    // restoring focus.
+    $("html").scrollTop(0);
```

Comment:
This seems OK but I think it'd be better to have it in `show` rather than `complete_rerender`, since the motivation is about the switch from another view?

Reply:
This is actually the correct place for it since it needs to be called before
`revive_current_focus`.